frontend/src/app/workflows/workflows.component.ts

```diff
@@ -0,0 +1,163 @@
+import {HttpClient} from '@angular/common/http';
+import {Component, HostListener, NgZone, OnInit, ViewChild} from '@angular/core';
+
+import {ToastService} from 'ng-bootstrap-ext';
+import {EditorConfiguration} from 'codemirror';
+import {PrivacyService} from '../privacy.service';
+import {LintService} from '../shared/lint.service';
+import {WorkflowsService} from './workflows.service';
+import {GenerateResult} from './model/GenerateResult';
+import {IOutputData, SplitComponent} from 'angular-split';
+import {environment} from '../../environments/environment';
+import {cmWorkflowsHint} from './model/helper/workflows-hint';
+
+@Component({
+  selector: 'app-workflows',
+  templateUrl: './workflows.component.html',
+  styleUrls: ['./workflows.component.scss']
+})
+export class WorkflowsComponent implements OnInit {
+  @ViewChild('split') split!: SplitComponent;
+
+  public content!: any;
+  public codemirrorOptions: EditorConfiguration | Record<string, any>;
+
+  public generateResult!: GenerateResult;
+
+  public currentExampleDesc: string | null = null;
+  public examplesList = ['Data Modelling', 'Microservices', 'Pages'];
+
+  public showIframeHider = false;
+  public newPageIndex: number = 0;
+  public currentDisplay: 'pages' | 'objects' | 'class' = 'pages';
+
+  public loading: boolean = false;
+
+  constructor(
+    private toastService: ToastService,
+    private zone: NgZone,
+    private fulibWorkflowsService: WorkflowsService,
+    private privacyService: PrivacyService,
+    private http: HttpClient,
+    private lintService: LintService,
+  ) {
+    // https://angular.io/api/core/NgZone
+    const generateHandler = () => this.zone.run(() => this.generate());
+
+    this.codemirrorOptions = {
+      lineNumbers: true,
+      mode: 'yaml',
+      extraKeys: {
+        'Ctrl-Space': 'autocomplete',
+        'Ctrl-S': generateHandler,
+        'Cmd-S': generateHandler,
+      },
+      autofocus: true,
+      tabSize: 2,
+      hintOptions: {
+        hint: cmWorkflowsHint,
+      },
+    };
+  }
+
+  ngOnInit() {
+    this.checkForStoredContent();
+  }
+
+  private checkForStoredContent() {
+    const storage = this.privacyService.getStorage('workflows');
+
+    if (storage && !this.currentExampleDesc) {
+      this.content = storage;
+      this.generate();
+    } else {
+      this.setExample(this.currentExampleDesc);
+    }
+  }
+
+  changeExampleContent(newExample: string | null) {
+    this.currentExampleDesc = newExample;
+    this.checkForStoredContent();
+  }
+
+  generate() {
+    if (!this.currentExampleDesc) {
+      this.privacyService.setStorage('workflows', this.content);
+    }
+
+    if (this.loading) return;
+
+    const newContent = this.lintService.lintYamlString(this.content);
+
+    if (!newContent) {
+      return;
+    } else {
+      this.content = newContent;
+    }
```

Comment:
Else is unnecessary due to early return

```suggestion
    }

    this.content = newContent;
```

Reply:
Was removed due to the removal of the lint service method